server/services/auth_service.py

```
@@ -50,30 +77,78 @@ async def signup(self, signup_data: SignupRequest) -> SignupResponse:
+                        "returnSecureToken": True
+                    }
+                )
+                
+                if response.status_code != 200:
+                    # Firebase returns 400 for invalid credentials
+                    # Don't reveal whether email exists - same error for both cases
+                    logger.warning(f"Failed login attempt for email: {login_data.email}")
+                    raise HTTPException(
+                        status_code=status.HTTP_401_UNAUTHORIZED,
+                        detail="Invalid email or password"
+                    )
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)